### DISCUSSION TITLE
configurable Base URL for the UI

### DISCUSSION THREAD

**Discussion by tim-roethig** [open-webui/open-webui#7122] [General]
### Discussed in https://github.com/open-webui/open-webui/discussions/3431

<div type='discussions-op-text'>

<sup>Originally posted by **tim-roethig-db** June 25, 2024</sup>
I want to run Open Web UI under a configurable Base URL. e.g. http://localhost:8080/MY_CUSTOM_BASE

It would be nice to have a env var for this.

I thought it might be possible to do it via the WEBUI_URL env var, but it seems to be mostly unused if I see this correctly. 
I also found [this code](https://github.com/open-webui/open-webui/blob/9e4dd4b86f77653b59dc1fa97fe3f72e8252b359/src/lib/constants.ts#L7), where the BASE_URL seems to be hard coded, maybe this can be altered easily.

Thanks for your time and the amazing project. Hope I didn't overlooked anything.</div>

Find here some tries of mine:
I initially thought it might be possible by doing some easy fixes, such as adding a root_path to the fastapis or adding a base path to the svelte.config.js as in this draft pr: https://github.com/tim-roethig-db/open-webui/pull/5

But it looks like this is not enough, since the svelte code contains a lot of hard coded absolute paths in goto, href and src. So I think quite a lot of changes in the svelte files might be necessary as i this draft pr: https://github.com/tim-roethig-db/open-webui/pull/4

Unfortunately both of these tries do not work. The best I achieved was to get the auth and the home screen to be served behind a base path.

**Comment by tjbck** [comment]
Out of scope of this project. It will never be supported officially, however, I'm open to accepting PRs!

**Reply by iSteeb** [reply]
[lmorin-inria](https://github.com/lmorin-inria) #1162 

seems like there's someone who has already worked on and solved this issue.  kinda weird you don't want to officially support it, but if you're open to PR's why was his rejected?

would be great to get this working

**Reply by gitaxin** [reply]
这个需求可以被支持吗？

**Reply by gitaxin** [reply]
有其他解决了这个问题吗？有没有仓库地址？

**Reply by jack9603301** [reply]
> 有其他解决了这个问题吗？有没有仓库地址？

注意使用英语

**Reply by jack9603301** [reply]
> [lmorin-inria](https://github.com/lmorin-inria) #1162 
> 
> seems like there's someone who has already worked on and solved this issue.  kinda weird you don't want to officially support it, but if you're open to PR's why was his rejected?
> 
> would be great to get this working

I looked at it briefly, and maybe it only takes one nginx.

**Reply by iSteeb** [reply]
> > [lmorin-inria](https://github.com/lmorin-inria) #1162
> > seems like there's someone who has already worked on and solved this issue.  kinda weird you don't want to officially support it, but if you're open to PR's why was his rejected?
> > would be great to get this working
> 
> I looked at it briefly, and maybe it only takes one nginx.

when i was using this project, nginx wasn’t enough, at least not with a ridiculous config full of rewrites for every explicit route
